File: Cpp/Day1/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s)
    {
        unordered_map<char,int>m;
        int l=0,maxlen=0;
        for(int i=0;i<s.size();i++)
        {
            int c=s[i];
            if(m.count(c)&&m[c]>=l)
            {
                l=m[c]+1;
            }
            m[c]=i;
            maxlen=max(maxlen,i-l+1);

        }
        return maxlen;
    }
};

File: Java/Day1-23.06.2025/LongestSubstringWithoutRepeatingCharacters.java
class Solution {
    public int lengthOfLongestSubstring(String s) 
    {
        Map<Character,Integer> map=new HashMap<>();
        
        int l=0, maxLen=0;
        for(int i=0;i<s.length();i++)
        {
            char c=s.charAt(i);
            if(map.containsKey(c) && map.get(c)>=l)
            {
                l=map.get(c)+1;
            }
                map.put(c,i);
            
            maxLen=Math.max(maxLen,i-l+1);
        }

        return maxLen;

        
    }
}
